common/db.go

```diff
@@ -269,14 +273,43 @@ func InitDB(dbPath string) (*DB, error) {
 	d.withdrawableUnbondingLocks = stmt
 
 	// Winning tickets prepared statements
-	stmt, err = db.Prepare("INSERT INTO winningTickets(sender, recipient, faceValue, winProb, senderNonce, recipientRand, recipientRandHash, sig, sessionID) VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)")
+	stmt, err = db.Prepare(`
+	INSERT INTO winningTickets(createdAt, sender, recipient, faceValue, winProb, senderNonce, recipientRand, recipientRandHash, sig, creationRound, creationRoundBlockHash)
+	VALUES(datetime('now'), :sender, :recipient, :faceValue, :winProb, :senderNonce, :recipientRand, :recipientRandHash, :sig, :creationRound, :creationRoundBlockHash)
+	`)
 	if err != nil {
 		glog.Error("Unable to prepare insertWinningTicket ", err)
 		d.Close()
 		return nil, err
 	}
 	d.insertWinningTicket = stmt
 
+	// Find latest ticket
+	stmt, err = db.Prepare("SELECT sender, recipient, faceValue, winProb, senderNonce, recipientRand, recipientRandHash, sig, creationRound, creationRoundBlockHash FROM winningTickets WHERE sender=? ORDER BY createdAt ASC LIMIT 1")
+	if err != nil {
+		glog.Error("Unable to prepare loadLatestTicket")
+		d.Close()
+		return nil, err
+	}
+	d.loadLatestTicket = stmt
+
+	stmt, err = db.Prepare("SELECT count(sig) FROM winningTickets WHERE sender=?")
+	if err != nil {
+		glog.Error("Unable to prepare loadWinningTicketCount")
+		d.Close()
+		return nil, err
+	}
+	d.winningTicketCount = stmt
+
+	// Remove latest ticket
+	stmt, err = db.Prepare("DELETE FROM winningTickets WHERE sender=? AND sig=?")
```

Comment:
In addition to persistence, the DB can be really useful as a forensic / exploratory tool - we lose that property if we're removing data from it.

For "deletion" can we set something like a "submittedAt" column?  For the latest tickets, `SELECT ... FROM winningTickets where submittedAt IS NULL` . When the ticket is "deleted", set submittedAt to the current timestamp. If it becomes too annoying to deal with the conditional, we can create a view table of non-submitted tickets.

(Later on we could also store things like transaction ids, confirmation block, etc... lots of interesting things we can put in the DB. This could also come in handy independently, eg if the tx needs to be re-submitted for whatever reason, such as a spike in gas prices.)

Reply:
I like storing redeemed ticket transaction hashes (metadata lookup can be done remotely using the hash) but wouldn't it be better to use a separate table for this ? 

While `ticketqueues`  are per-sender this table isn't and having this table not grow too large while still having sender indexing seems like a nice property to have in terms of efficiency.

Reply:
>  wouldn't it be better to use a separate table for this 

It could be a separate table - depends on what we want to do with it. Eg, track re-submissions? Then maybe an independent table would make sense, with one row per submission. It's kind of nice to have the data alongside other ticket parameters though, or maybe the final tx hash at least. Makes common types of queries easier and we don't have to worry about cross-table consistency.

> having this table not grow too large while still having sender indexing seems like a nice property to have in terms of efficiency.

I wouldn't be too concerned about the table size. The data is valuable in itself, and with effective indexing, lookups will be very fast. If the size really becomes an issue later on, we can always offer tools for pruning the table. Until then, it seems like a loss to be deleting this data.

Reply:
I believe the broker handles transaction retries already so tickets should only be removed from the database once they are succesfully redeemed, otherwise they are readded to the DB. 

I'm wondering if there's other alternatives possible to serve as an archive for forensic tools such as the subgraph. The only data currently we wouldn't have access to through the event indexing is the `pricePerPixel` for specific tickets. It's part of the HMAC signature but not part of any on-chain data but we could add it to the auxData() for exactly this purpose down the line.

Reply:
> I believe the broker handles transaction retries already

There currently is no redemption tx retry/replace logic for the case where a redemption tx sits pending for awhile and then `CheckTx()` times out. Adding such logic is tracked by #654.

Reply:
Should we consider #654 in scope for this epic?

Reply:
Can we still consider marking submitted rows as such, rather than deleting them from the table outright? There is really very little reason to delete data here.

Reply:
Regarding #654 I think we can keep it out of scope since tickets are essentially retried because they are added back to the queue. 

In terms of not deleting winning tickets would the storage growth be a worthwhile tradeoff to make considering once a ticket has been redeemed we also have historical data available through the subgraph or archive ETH nodes ?

Reply:
> Should we consider #654 in scope for this epic?

Let's keep #654 outside of the scope of this epic since the ticket queue can already handle retries for tickets that have not been successfully submitted. A solution for #654 may be useful for handling tx replacements with a gas price bump if the tx is not mined within some time window, but that can be addressed later.

> Would the storage growth be a worthwhile tradeoff to make considering once a ticket has been redeemed we also have historical data available through the subgraph or archive ETH nodes ?

I can see the benefits of keeping ticket data even after redemption based on the discussion thus far i.e. forensics/analysis without dependence on external tools, treating tickets as valuable data that shouldn't be deleted by default, etc.

Another benefit I see is if the node crashes in between removing a ticket from the queue and re-adding the ticket to the queue then the node can still redeem the ticket when it restarts. An example scenario might be if the node is waiting for the redemption tx to be mined when gas prices are high, the node crashes for some reason and the redemption tx is never mined because gas prices keep rising.

The main disadvantage of keeping ticket data even after redemption seems to be storage growth. But, storage growth should be manageable (if each ticket is ~284 bytes then 100000 winning tickets would be 28.4 MB) and as was mentioned earlier in the discussion we can add an opt-in pruning routine to delete really old tickets if needed. 

Given the advantages of keeping ticket data even after redemption and the manageable disadvantage of keeping ticket data, I think keeping ticket data makes sense.

Reply:
fixed in d69225c ab7363e